docs/building-blocks/extending-templates/collector.md

```diff
@@ -0,0 +1,308 @@
+---
+title: Implement a property transformer and collector in an Azure Resource Manager template
+description: Describes how to implement a property transformer and collector in an Azure Resource Manager template
+author: petertay
+ms.date: 06/09/2017
+
+---
+
+# Implement a property transformer and collector in an Azure Resource Manager template
+
+In [use an object as a parameter in an Azure Resource Manager template][objects-as-parameters], you learned how to store resource property values in an object and apply them to a resource during deployment. While this is a very useful way to manage your parameters, it still requires you to map the object's properties to resource properties each time you use it in your template.
+
+To work around this, you can implement a property transform and collector that iterates your object array and transforms it into the JSON schema expected by the resource.
+
+> [!IMPORTANT]
+> This approach requires that you have a deep understanding of Resource Manager templates and functions.
+
+Let's take a look at how we can implement a property collector and transformer with an example that deploys a [network security group (NSG)][nsg]. The diagram below shows the relationship between our templates and our resources within those templates:
+
+![property collector and transformer architecture](../_images/collector-transformer.png)
+
+Our **calling template** includes two resources:
+* a template link that invokes our **collector template**.
+* an NSG.
```

Comment:
Is that an NSG resource? As in the exact resource shape expected by the RP? If so, what are we transforming? If not, can we be more specific about what we are doing here?